cpp/oneapi/dal/data/undirected_adjacency_array_graph.cpp

```diff
@@ -135,35 +102,37 @@ void convert_to_csr_impl(const edge_list<vertex_type<G>> &edges, G &g) {
     }
 
     layout->_vertex_count = max_id + 1;
-    layout->_degrees      = array<vertex_t>::zeros(layout->_vertex_count);
+    int_t *degrees        = (int_t *)malloc(layout->_vertex_count * sizeof(int_t));
+    for (int_t u = 0; u < layout->_vertex_count; ++u) {
+        degrees[u] = 0;
+    }
 
     for (auto edge : edges) {
-        layout->_degrees[edge.first]++;
-        layout->_degrees[edge.second]++;
+        degrees[edge.first]++;
+        degrees[edge.second]++;
     }
 
-    layout->_vertex_neighbors = array<vertex_t>::empty(layout->_vertex_count + 1);
-    auto _rows                = layout->_vertex_neighbors.get_mutable_data();
-    int_t total_sum_degrees   = 0;
-    _rows[0]                  = total_sum_degrees;
+    layout->_vertexes.resize(layout->_vertex_count + 1);
+    auto _rows              = layout->_vertexes.data();
+    int_t total_sum_degrees = 0;
+    _rows[0]                = total_sum_degrees;
 
-    for (std::int64_t i = 0; i < layout->_vertex_count; ++i) {
-        total_sum_degrees += layout->_degrees[i];
+    for (int_t i = 0; i < layout->_vertex_count; ++i) {
+        total_sum_degrees += degrees[i];
         _rows[i + 1] = total_sum_degrees;
     }
 
-    layout->_edge_offsets = array<edge_t>::empty(_rows[layout->_vertex_count] + 1);
-    auto _cols            = layout->_edge_offsets.get_mutable_data();
-    auto offests(layout->_vertex_neighbors);
+    free(degrees);
+    layout->_edges.resize(_rows[layout->_vertex_count] + 1);
```

Comment:
general question - I assume this resize comes from the use of std::vector as a container. If this is the case, we put into the lib another rt depedence, and the question why this is not based in headers/detail namespace?